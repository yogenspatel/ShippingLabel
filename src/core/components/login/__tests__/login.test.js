/* eslint-disable no-undef */

import React from "react";
import ReactDOM from "react-dom";
import { Login } from "../index";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

describe("Login Component renders correctly", () => {
  it("has rendered without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Login />, div);
  });
  it("has called snapshot with Logged in component", () => {
    const props = {
        username: 'test',
        password: 'test',
        handleOnChange: jest.fn,
        handleOnSubmit: jest.fn,
        isLoggedIn: false,
        error: false
    }
    const tree = shallow(<Login {...props} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("has called componentDidMount method", () => {
    const spy = jest.spyOn(Login.prototype, 'componentDidMount');
    const props = {
        username: 'test',
        password: 'test',
        handleOnChange: jest.fn,
        handleOnSubmit: jest.fn,
        isLoggedIn: false,
        error: false
    }
    const wrapper = shallow(<Login {...props} />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
  it("has a snapshot when there is an error", () => {
    const props = {
        username: 'test',
        password: 'test',
        handleOnChange: jest.fn,
        handleOnSubmit: jest.fn,
        isLoggedIn: false,
        error: true
    }
    const tree = shallow(<Login {...props} />);
    expect(tree.find(".alert-danger").exists()).toBeTruthy();
  });
  it("has form submit button", () => {
    const props = {
        username: 'test',
        password: 'test',
        handleOnChange: jest.fn,
        handleOnSubmit: jest.fn,
        isLoggedIn: false,
        error: false
    }
    const tree = shallow(<Login {...props} />);
    expect(tree.find(".form-submit").exists()).toBeTruthy();

  });
  it("has logged in", () => {
    const props = {
        username: 'test',
        password: 'test',
        handleOnChange: jest.fn,
        handleOnSubmit: jest.fn,
        isLoggedIn: true,
        error: false
    }
    const tree = mount(<Login {...props} />);
    expect(tree.find('.progress-bar').exists()).toBeTruthy();
  });
});
