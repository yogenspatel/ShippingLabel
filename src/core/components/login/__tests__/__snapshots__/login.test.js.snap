// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly has called snapshot with Logged in component 1`] = `
<div
  className="card"
>
  <div
    className="card-header text-center h3"
  >
    Login (Username = test | Password = test)
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group input-group-lg mb-3 ml-5 mt-5 pr-5"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          Username
        </span>
      </div>
      <input
        autoComplete="new-username"
        className="form-control"
        name="username"
        onChange={[Function]}
        type="text"
        value="test"
      />
    </div>
    <div
      className="input-group input-group-lg mb-3 ml-5 pr-5"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          Password
        </span>
      </div>
      <input
        autoComplete="new-password"
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
        value="test"
      />
    </div>
    <div
      className="text-center mb-3"
    >
      <input
        className="btn btn-primary form-submit"
        type="submit"
        value="Login"
      />
    </div>
  </form>
</div>
`;
